src/app/core/learning-object-module/file/file.service.ts

```diff
@@ -53,6 +59,198 @@ export class FileService {
       .then((blob) => window.URL.createObjectURL(blob));
   }
 
+  /**
+   * Makes request to update file description
+   *
+   * @param {string} authorUsername
+   * @param {string} objectId
+   * @param {string} fileId
+   * @param {string} description
+   * @returns {Promise<any>}
+   * @memberof LearningObjectService
+   */
+  updateFileDescription(
+    authorUsername: string,
+    objectId: string,
+    fileId: string,
+    description: string
+  ): Promise<any> {
+    const route = FILE_ROUTES.UPDATE_FILE(
+      authorUsername,
+      objectId,
+      fileId
+    );
+    return this.http
+      .patch(
+        route,
+        { description },
+        { headers: this.headers, withCredentials: true, responseType: 'text' }
+      )
+      .pipe(
+
+        catchError(this.handleError)
+      )
+      .toPromise();
+  }
+
+  // Updates the README of a learning object
+  async updateReadme(id: string): Promise<any> {
+    return await this.http.patch(
+      FILE_ROUTES.UPDATE_PDF(id),
+      {},
+      {
+        headers: this.headers,
+        withCredentials: true,
+        responseType: 'text'
+      }
+    )
+      .pipe(
+
+        catchError(this.handleError)
+      ).toPromise();
+  }
+
+  /**
+   * Fetches Learning Object's Materials
+   *
+   * @param {string} authorUsername
+   * @param {string} objectId
+   * @param {string} description
+   * @returns {Promise<any>}
+   * @memberof LearningObjectService
+   */
+    getMaterials(objectId: string): Promise<any> {
+      const route = FILE_ROUTES.GET_MATERIALS(objectId, 'unreleased');
+      return this.http.get(route, { withCredentials: true })
+        .pipe(
+          catchError(this.handleError)
+        )
+        .toPromise();
+    }
+
+  /**
+   * Adds file meta to a Learning Object's materials
+   * Adding files are handled by a job queue to avoid sending too large of a payload to the server
+   *
+   * @param {string} authorUsername [The Learning Object's author's username]
+   * @param {string} objectId [The Id of the Learning Object]
+   * @param {FileUploadMeta[]} files [List of file meta to be added]
+   * @returns {Promise<string[]>}
+   * @memberof LearningObjectService
+   */
+  addFileMeta({
+    objectId,
+    files
+  }: {
+      objectId: string;
+      files: FileUploadMeta[];
+    }): Promise<string[]> {
+      const route = FILE_ROUTES.UPLOAD_FILE_META(objectId);
+      return this.handleFileMetaRequests(files, route);
+    }
```

Comment:
Nit: If nothing else uses handleFileMetaRequests, can we just combine the functions so it is more clear to what is happening?

Reply:
+1